### PR TITLE
Clean Data Script Initial Pull Request #1

### PR DESCRIPTION
**Pull request by morganjacobus**
This is the initial pull request for the bulk of the clean data script. Most of the data cleaning tasks have been incorporated into the code and in the Bus Stop Census Corrections Log. A few cleaning tasks remain to be added to the script. These can be found on the Bus Stop Census Corrections Log.

### CODE REVIEW

File: code/raw_code/01_clean_data.R [hunk 1]
```diff
@@ -195,72 +213,123 @@
...
+                              select(Record_ID, Stop_ID, Timestamp, Email_Masked, Main_Street, Nearest_Landmark, Routes, Direction,
+                                     Seating, Shelter, Trash_Can, Litter, Grafitti, Overflow, Dirty_Seating, Other, Line_of_Sight,
+                                     Route_Number, Route_Schedule, Route_Map, Customer_Service, None_Of_The_Above, Wayfinding_Accessibility,
+                                     Lighting, Sidewalk, Obstacles, Obstacle_Desc, Boarding_Area, Main_Street_Crosswalk, Cross_Street_Crosswalk, 
+                                     Worn_Faded, No_Crosswalk, Traffic_Light, Curb_Cuts, Crosswalk_Signals, Crossing_Audio, Tactile_Guide, Informal_Pathways,
+                                     Compete_For_Seat, Cross_Midblock, Catch_The_Bus, Dangerous_Motorists, First_Visit, Regular_User_None,
+                                     On_Site_Survey, Additional_Comments)
+  
+  ###################### SPREAD DATA FOR SELECT ALL THAT APPLY RESPONSES #########################
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R:254]: This comment doesn't make sense.

File: code/raw_code/01_clean_data.R [hunk 2]
```diff
@@ -11,26 +11,29 @@
...
   # You must have a Google Account to use the package as it will ask you to authorize the package to access your Google Sheets.
   # There is a limit on how many reads can be made through the Google Sheets API. 
   # Import Corrections Sheet
-  corrections_df <- read_sheet("https://docs.google.com/spreadsheets/d/1n8RErNhOXrNptVyul5rTuE-ZTdSzkljB0cCAV8hBJkw/edit?usp=sharing",
-                               sheet = "Corrections by Record ID",
-                               col_names = TRUE,
-                               col_types = "icccccc",
-                               na = "") 
+  corrections_df <- read_sheet(
+                         ss        = "https://docs.google.com/spreadsheets/d/1n8RErNhOXrNptVyul5rTuE-ZTdSzkljB0cCAV8hBJkw/edit?usp=sharing",
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Move instances of the spreadsheet URL into a variable, place that variable below the comment on line 12.

File: code/raw_code/01_clean_data.R [hunk 3]
```diff
@@ -1,5 +1,5 @@
   
-  ########################### READ ME ###########################
+  ###################### INTRODUCTION TO THE CLEAN DATA SCRIPT ######################### 
   # A table of all cleaning tasks performed, their purpose, and their location are contained within the Bus Stop Corrections Log at the following linK:
   # https://docs.google.com/spreadsheets/d/1n8RErNhOXrNptVyul5rTuE-ZTdSzkljB0cCAV8hBJkw/edit?usp=sharing
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: If creating a variable, mention the URL in the variable instead?

File: code/raw_code/01_clean_data.R [hunk 4]
```diff
@@ -11,26 +11,29 @@
...
+                    ) 
   
   # Import Missing Question Data
-  missing_questions_populated_df <- read_sheet("https://docs.google.com/spreadsheets/d/1n8RErNhOXrNptVyul5rTuE-ZTdSzkljB0cCAV8hBJkw/edit?usp=sharing",
-                                    sheet = "Missing Question Records",
-                                    col_names = TRUE,
-                                    na = "") %>%
-                                    select(Record_ID, Sidewalk_New, Obstacles_New, Obstacle_Desc_New, Boarding_Area_New, Crosswalk_Features_New) # Selects only the fields that were filled in
+  missing_questions_populated_df <- read_sheet(
+                                          ss = "https://docs.google.com/spreadsheets/d/1n8RErNhOXrNptVyul5rTuE-ZTdSzkljB0cCAV8hBJkw/edit?usp=sharing",
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Move instances of the spreadsheet URL into a variable, place that variable below the comment on line 12.

File: code/raw_code/01_clean_data.R [hunk 5]
```diff
@@ -11,26 +11,29 @@
...
+                                          sheet = "Missing Question Records",
+                                          col_names = TRUE,
+                                          na = "") %>%
+                                    select(Record_ID, Sidewalk_New, Obstacles_New, 
+                                           Obstacle_Desc_New, Boarding_Area_New, Crosswalk_Features_New) # Selects only the fields that were filled in
   
   # Import Duplicate Records By Stop ID and Fields Data
   duplicate_records_reconciled_df <- read_sheet("https://docs.google.com/spreadsheets/d/1n8RErNhOXrNptVyul5rTuE-ZTdSzkljB0cCAV8hBJkw/edit?usp=sharing",
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Move instances of the spreadsheet URL into a variable, place that variable below the comment on line 12.

File: code/raw_code/01_clean_data.R [hunk 6]
```diff
@@ -59,114 +62,138 @@
...
+                          )
+  
+  ######################### FILL IN MISSING QUESTION DATA #########################
+  # Questions corresponding to Sidewalk, Obstacle, Obstacles_Desc, Boarding_Area, and Crosswalk_Features were added shortly after
+  # the Bus Stop Census survey was launched. This resulted in over 100 surveys being submitted without answers for these fields.
+  # The following code identifies these records and exports them to a csv for the answers to be filled in. | Data Cleaning Task #10
   
-  ## FILL IN MISSING QUESTION DATA ----------------------------------------------------------------------------------------------
   # Create data.frame of records with missing data for Sidewalk, Obstacle, Obstacle_Desc, Boarding_Area, and Crosswalk_Features questions
   'Apporximately XX days after the start of data collection, four new questions were added:
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Typo "Approximately..."

File: code/raw_code/01_clean_data.R [hunk 7]
```diff
@@ -59,114 +62,138 @@
...
-                                   Wayfinding_Accessibility = ifelse(!is.na(Wayfinding_Accessibility_Reconciled), Wayfinding_Accessibility_Reconciled, Wayfinding_Accessibility),
-                                   Sidewalk = ifelse(!is.na(Sidewalk_Reconciled), Sidewalk_Reconciled, Sidewalk),
-                                   Boarding_Area = ifelse(!is.na(Boarding_Area_Reconciled), Boarding_Area_Reconciled, Boarding_Area),
-                                   Crosswalks = ifelse(!is.na(Crosswalks_Reconciled), Crosswalks_Reconciled, Crosswalks),
-                                   Crosswalk_Features = ifelse(!is.na(Crosswalk_Features_Reconciled), Crosswalk_Features_Reconciled, Crosswalk_Features)) %>%
-                            select(-Seating_Reconciled, -Shelter_Reconciled, -Trash_Can_Reconciled,
-                                   -Wayfinding_Reconciled, -Wayfinding_Accessibility_Reconciled, -Sidewalk_Reconciled,
-                                   -Boarding_Area_Reconciled, -Crosswalks_Reconciled, -Crosswalk_Features_Reconciled)
+                              left_join(duplicate_records_reconciled_df, by = c("Record_ID")) %>%
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: The extra indents are not consistent with above assignments to `raw_df_clean_deduped`, are they needed?

File: code/raw_code/01_clean_data.R [hunk 8]
```diff
@@ -195,72 +223,137 @@
...
+                                     Dirty_Seating = ifelse(grepl("Dirty seating area", Cleanliness, fixed = TRUE), "Yes", "No"),
+                                     Other = ifelse(grepl("Other", Cleanliness, fixed = TRUE), "Yes", "No")) %>%
+                              mutate(Route_Number = ifelse(grepl("Route Numbers", Wayfinding, fixed = TRUE), "Yes", "No"),
+                                     Route_Schedule = ifelse(grepl("Route Schedule", Wayfinding, fixed = TRUE), "Yes", "No"),
+                                     Route_Map = ifelse(grepl("Route Map", Wayfinding, fixed = TRUE), "Yes","No"),
+                                     Customer_Service = ifelse(grepl("Customer Service Information", Wayfinding, fixed = TRUE), "Yes", "No"),
+                                     None_Of_The_Above = ifelse(grepl("None of the Above", Wayfinding, fixed = TRUE), "Yes", "No")) %>%
+                              mutate(Main_Street_Crosswalk = ifelse(grepl("Yes, on the main street", Crosswalks, fixed = TRUE), "Yes", "No"),
+                                     Cross_Street_Crosswalk = ifelse(grepl("Yes, on the cross street",Crosswalks, fixed = TRUE), "Yes", "No"),
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Insert space before `Crosswalks`.

File: code/raw_code/01_clean_data.R [hunk 9]
```diff
@@ -195,72 +223,137 @@
...
+                                     Worn_Faded = ifelse(grepl("Yes, and crosswalk paint is faded or worn away", Crosswalks, fixed = TRUE), "Yes","No"),
+                                     No_Crosswalk = ifelse(grepl("No, no painted crosswalk within 100 feet", Crosswalks, fixed = TRUE), "Yes", "No")) %>%
+                              mutate(Traffic_Light = ifelse(grepl("Traffic_Light", Crosswalk_Features, fixed = TRUE), "Yes", "No"),
+                                     Curb_Cuts = ifelse(grepl("Curb cuts for wheelchairs",Crosswalk_Features, fixed = TRUE), "Yes", "No"),
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Insert space before `Crosswalk_Features`.

File: code/raw_code/01_clean_data.R [hunk 10]
```diff
@@ -195,72 +223,137 @@
...
+                                     Crosswalk_Signals = ifelse(grepl("Crosswalk signals with push buttons", Crosswalk_Features, fixed = TRUE), "Yes","No"),
+                                     Crossing_Audio = ifelse(grepl("Crossing audio overlays for the visually impaired", Crosswalk_Features, fixed = TRUE), "Yes", "No"),
+                                     Tactile_Guide = ifelse(grepl("Tactile guide strips for the visually impaired", Crosswalk_Features, fixed = TRUE), "Yes", "No")) %>%
+                              mutate(Informal_Pathways = ifelse(grepl("Pedestrians using informal pathways where sidewalks do not exist", Behavior, fixed = TRUE), "Yes", "No"),
+                                     Compete_For_Seat = ifelse(grepl("Pedestrians competing for seating at the bus stop",Behavior, fixed = TRUE), "Yes", "No"),
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Insert space before `Behavior`.

File: code/raw_code/01_clean_data.R [hunk 11]
```diff
@@ -195,72 +223,137 @@
...
+                                     Cross_Midblock = ifelse(grepl("Pedestrians crossing the roadway at midblock locations", Behavior, fixed = TRUE), "Yes","No"),
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Insert space before `"No"`.

File: code/raw_code/01_clean_data.R [hunk 12]
```diff
@@ -195,72 +223,137 @@
...
+                                     None_Of_The_Above = ifelse(grepl("None of the Above", Wayfinding, fixed = TRUE), "Yes", "No")) %>%
+                              mutate(Main_Street_Crosswalk = ifelse(grepl("Yes, on the main street", Crosswalks, fixed = TRUE), "Yes", "No"),
+                                     Cross_Street_Crosswalk = ifelse(grepl("Yes, on the cross street",Crosswalks, fixed = TRUE), "Yes", "No"),
+                                     Worn_Faded = ifelse(grepl("Yes, and crosswalk paint is faded or worn away", Crosswalks, fixed = TRUE), "Yes","No"),
+                                     No_Crosswalk = ifelse(grepl("No, no painted crosswalk within 100 feet", Crosswalks, fixed = TRUE), "Yes", "No")) %>%
+                              mutate(Traffic_Light = ifelse(grepl("Traffic_Light", Crosswalk_Features, fixed = TRUE), "Yes", "No"),
+                                     Curb_Cuts = ifelse(grepl("Curb cuts for wheelchairs",Crosswalk_Features, fixed = TRUE), "Yes", "No"),
+                                     Crosswalk_Signals = ifelse(grepl("Crosswalk signals with push buttons", Crosswalk_Features, fixed = TRUE), "Yes","No"),
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Insert space before `"No"`.

File: code/raw_code/01_clean_data.R [hunk 13]
```diff
@@ -195,72 +223,137 @@
...
+                              mutate(Route_Number = ifelse(grepl("Route Numbers", Wayfinding, fixed = TRUE), "Yes", "No"),
+                                     Route_Schedule = ifelse(grepl("Route Schedule", Wayfinding, fixed = TRUE), "Yes", "No"),
+                                     Route_Map = ifelse(grepl("Route Map", Wayfinding, fixed = TRUE), "Yes","No"),
+                                     Customer_Service = ifelse(grepl("Customer Service Information", Wayfinding, fixed = TRUE), "Yes", "No"),
+                                     None_Of_The_Above = ifelse(grepl("None of the Above", Wayfinding, fixed = TRUE), "Yes", "No")) %>%
+                              mutate(Main_Street_Crosswalk = ifelse(grepl("Yes, on the main street", Crosswalks, fixed = TRUE), "Yes", "No"),
+                                     Cross_Street_Crosswalk = ifelse(grepl("Yes, on the cross street",Crosswalks, fixed = TRUE), "Yes", "No"),
+                                     Worn_Faded = ifelse(grepl("Yes, and crosswalk paint is faded or worn away", Crosswalks, fixed = TRUE), "Yes","No"),
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Insert space before `"No"`.

File: code/raw_code/01_clean_data.R [hunk 14]
```diff
@@ -195,72 +223,137 @@
...
+  raw_df_clean_deduped <- raw_df_clean_deduped %>% 
+                              mutate(Litter = ifelse(grepl("Litter at the stop", Cleanliness, fixed = TRUE), "Yes", "No"),
+                                     Grafitti = ifelse(grepl("Graffiti (unauthorized) or tagging on bus stop amenities",Cleanliness, fixed = TRUE), "Yes", "No"),
+                                     Overflow = ifelse(grepl("Overflowing or poorly maintained trash can", Cleanliness, fixed = TRUE), "Yes","No"),
+                                     Dirty_Seating = ifelse(grepl("Dirty seating area", Cleanliness, fixed = TRUE), "Yes", "No"),
+                                     Other = ifelse(grepl("Other", Cleanliness, fixed = TRUE), "Yes", "No")) %>%
+                              mutate(Route_Number = ifelse(grepl("Route Numbers", Wayfinding, fixed = TRUE), "Yes", "No"),
+                                     Route_Schedule = ifelse(grepl("Route Schedule", Wayfinding, fixed = TRUE), "Yes", "No"),
+                                     Route_Map = ifelse(grepl("Route Map", Wayfinding, fixed = TRUE), "Yes","No"),
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Insert space before `"No"`.

File: code/raw_code/01_clean_data.R [hunk 15]
```diff
@@ -195,72 +223,137 @@
...
-    select(-main_street_or_station, -cross_street) # Remove helper columns
-  
-  #Data Output
-  # write.csv(, file = here("data", "tidy_data", "pre_processed_survey_df.csv"))
+  ###################### SPREAD DATA FOR SELECT ALL THAT APPLY RESPONSES ######################### 
+  
+  # Separate Select All That Apply Question Fields With Multiple Answers Into One Field Per Answer | Data Cleaning Task #14
+  raw_df_clean_deduped <- raw_df_clean_deduped %>% 
+                              mutate(Litter = ifelse(grepl("Litter at the stop", Cleanliness, fixed = TRUE), "Yes", "No"),
+                                     Grafitti = ifelse(grepl("Graffiti (unauthorized) or tagging on bus stop amenities",Cleanliness, fixed = TRUE), "Yes", "No"),
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Insert space before `Cleanliness`.

File: code/raw_code/01_clean_data.R [hunk 16]
```diff
@@ -195,72 +223,137 @@
...
+                                     Overflow = ifelse(grepl("Overflowing or poorly maintained trash can", Cleanliness, fixed = TRUE), "Yes","No"),
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Insert space before `"No"`.

File: code/raw_code/01_clean_data.R [hunk 17]
```diff
@@ -195,72 +223,137 @@
...
+  wayfinding_verification <- raw_df_clean_deduped %>%
+                                mutate(Flag = ifelse(str_count(paste(Route_Number, Route_Schedule, Route_Map, sep = ""), "Yes") > 0 &
+                                                     Wayfinding_Accessibility == "No wayfinding information present",
+                                                   "FLAG", "")) %>%
+                                dplyr::filter(Flag == "FLAG")
+  #write.csv(wayfinding_verification, here("data", "raw_data", "validation", "Wayfinding and No Wayfinding Responses.csv"), row.names = FALSE)
+  
+  # Identify and Export Records where the options for Main Street, Cross Street, or Worn or Faded Crosswalk are selected along with the "No, there is no crosswalk within 100 feet" selection 
+  crosswalk_verification_pt_one <- raw_df_clean_deduped %>% 
+                                    mutate(Flag = ifelse(str_count(paste(Main_Street_Crosswalk, Cross_Street_Crosswalk,  Worn_Faded, sep = ""), "Yes") > 0 & No_Crosswalk == "Yes",
```
Comment (QYAN24) [inline code/raw_code/01_clean_data.R]: Delete extra space before `Worn_Faded`.

**Review by QYAN24 — APPROVED** [review]
Sorry for the delayed review, looks good, please see formatting comments and extract the spreadhseet URL to a variable.

File: code/raw_code/01_clean_data.R (hunk 17 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Space removed

File: code/raw_code/01_clean_data.R (hunk 16 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Inserted spaces before `"No"`

File: code/raw_code/01_clean_data.R (hunk 15 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Inserted

File: code/raw_code/01_clean_data.R (hunk 14 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Inserted

File: code/raw_code/01_clean_data.R (hunk 13 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Inserted

File: code/raw_code/01_clean_data.R (hunk 12 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Inserted

File: code/raw_code/01_clean_data.R (hunk 11 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Inserted

File: code/raw_code/01_clean_data.R (hunk 10 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Inserted

File: code/raw_code/01_clean_data.R (hunk 9 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Inserted

File: code/raw_code/01_clean_data.R (hunk 8 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Inserted

File: code/raw_code/01_clean_data.R (hunk 7 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Updated indentation for all dataframe assignments to be uniform

File: code/raw_code/01_clean_data.R (hunk 6 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Corrected spelling

File: code/raw_code/01_clean_data.R (hunk 5 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Created variable for URL and referenced in the function parameter calls

File: code/raw_code/01_clean_data.R (hunk 4 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Created variable for URL and referenced in the function parameter calls

File: code/raw_code/01_clean_data.R (hunk 3 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Created variable for URL and referenced in the comment

File: code/raw_code/01_clean_data.R (hunk 2 above)
Comment (morganjacobus) [reply code/raw_code/01_clean_data.R]: Created variable for URL and referenced in the function parameter calls

**Pull request merged**